waku/waku_sync/protocol.nim

```diff
@@ -54,25 +54,32 @@ proc serverReconciliation(self: WakuSync, message: seq[byte]): Result[seq[byte],
   ok(payload)
 
 proc clientReconciliation(
-  self: WakuSync, message: seq[byte],
+  self: WakuSync, negentropy: pointer, message: seq[byte],
   haveHashes: var seq[WakuMessageHash],
   needHashes: var seq[WakuMessageHash],
   ): Result[Option[seq[byte]], string] =
-  let payload: seq[byte] = negentropyClientReconcile(self.negentropy, message, haveHashes, needHashes)
+  let payload: seq[byte] = negentropyClientReconcile(negentropy, message, haveHashes, needHashes)
   ok(some(payload))
 
-proc intitialization(self: WakuSync): Future[Result[seq[byte], string]] {.async.} =
-  let payload: seq[byte] = negentropyInitiate(self.negentropy)
+proc intitialization(self: WakuSync, negentropy: pointer): Future[Result[seq[byte], string]] {.async.} =  
+  let payload: seq[byte] = negentropyInitiate(negentropy)
   info "initialized negentropy ", value=payload
 
   ok(payload)
 
+#[ template freeNegentropyAndReturn(negentropy: pointer, ret: untyped) =
+  negentropyFree(negentropy)
+  return ret ]#
+
 proc request(self: WakuSync, conn: Connection): Future[Result[seq[WakuMessageHash], string]] {.async, gcsafe.} =
-  let request: seq[byte] = (await self.intitialization()).valueOr:
+  let negentropy = negentropyNew(self.storage, DefaultFrameSize)
+  let request: seq[byte] = (await self.intitialization(negentropy)).valueOr:
+    negentropyFree(negentropy)
```

Comment:
I don't understand why do we need to free negentropy every request?

`negentropyFree(negentropy)` implie removing the storage as well no?

Reply:
Explained above the need for it. 
negentropy freeing doesn't remove storage. If you notice the way the negentropy API is implemented. 
Storage is for storing hashes and negentropy is for the protocol i.e doing actual sync between client and server.